documentation_quality_analysis/analyze_library/lib_analysis_service.py

```diff
@@ -0,0 +1,174 @@
+from analyze_library.metric_calculation.navigability.navigability import get_navigability_score
+from analyze_library.signature_matcher.python_doc_api_to_example_signature_matcher import \
+    match_examples_to_doc_api
+from analyze_library.signature_matcher.python_src_api_to_doc_api_matcher import \
+    match_doc_api_to_src_api
+from analyze_library.source_parser.source_code_parser import \
+    get_methods_and_classes_from_source_code
+from analyze_library.source_parser.source_util import clone_repo
+from analyze_library.stats.stats import get_stats_ex_per_api, get_stats_api_per_example, \
+    write_stats_to_file, write_doc_api_to_csv, write_examples_to_csv, get_stats_ex_per_methods_and_classes
+from analyze_library.metric_calculation.text_readability.text_readability import get_text_readability
+from analyze_library.metric_calculation.metrics import Metrics
+
+depth_map = {
+    'requests': 2,
+    'pandas': 3,
+    'graphql compiler': 3,
+    'collections': 0,
+    'flask': 2,
+    'nltk': 2,
+    'numba': 1,
+    'numpy': 3,
+    'gitpython': 2,
+    'pyppeteer': 1,
+    'python-socketio': 1,
+    'scanpy': 2,
+    'bionumpy': 1,
+    'wikidata': 1
+}
+
+
+def analyze_library(language, library_name, doc_url, gh_url):
+    depth = get_depth(library_name)
+    # os.chdir(ROOT_DIR)
+
+    repo_path = clone_repo(gh_url, True)
+    # repo_path = 'cloned_repos/' + library_name.strip()
+    print("Done cloning")
+
+    mapped_source_api: dict[str, List[Signature]] = get_methods_and_classes_from_source_code(
+        repo_name=library_name, repo_path=repo_path)
+
+    doc_pages: List[DocPage] = get_all_webpages(doc_url, depth)
+
+    doc_apis: List[Union[Signature, None]] = get_functions_and_classes_from_doc_api_ref(doc_pages)
+
+    matched_signatures, unmatched_signatures = match_doc_api_to_src_api(library_name, mapped_source_api,
+                                                                        doc_apis=doc_apis)
+
+    doc_code_examples: List[DocCodeExample] = get_functions_and_classes_from_doc_examples(doc_pages)
+
+    matched_methods: List[MatchedCall] = match_examples_to_doc_api(library_name, doc_code_examples, doc_apis)
+
+    stats_example_per_api = get_stats_ex_per_api(doc_apis, matched_methods)
+    stats_example_per_method, stats_example_per_class = get_stats_ex_per_methods_and_classes(doc_apis, matched_methods)
+
+    stats_api_per_example = get_stats_api_per_example(matched_methods, doc_code_examples)
+
+    # write_stats_to_file(doc_code_examples, stats_example_per_api, stats_api_per_example, library_name)
+    # write_doc_api_to_csv(doc_apis, library_name)
+    # write_examples_to_csv(doc_code_examples, library_name)
+
+    print(f'Done analyzing {library_name}')
+
+
+def get_depth(lib_name):
+    lib_name = lib_name.lower()
+    if lib_name in depth_map:
+        return depth_map[lib_name]
+    else:
+        return 2
+
+
+if __name__ == '__main__':
+    analyze_library("python", "requests",
```

Comment:
we may eventually want to make this configurable somehow. For example, an input json file with the libraries to process?